.github/templates/build-container/action.yaml

```diff
@@ -0,0 +1,36 @@
+name: build
+
+inputs:
+  dockerfile:
+    required: true
+  image:
+    required: true
+  registry:
+    required: true
+  user:
+    required: true
+  secret:
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout the repository
```

Comment:
I don't think checkout here is necessary, as this is an action and may not always be checking out master, I would leave the checkout to the workflow itself

Reply:
> * with a matrix step you can create a matrix with all the projects the need to be build, so u don't need to create individual workflows for each project

Strategy with matrix will not work as I need to watch for specific `on` conditions to trigger the jobs. And only one `on` is allowed per workflow, AFAIK.

Reply:
You can certainly build a matrix which includes selects only projects that contain changes, but we can do that in a separate PR.

Reply:
Using `on`? Interesting. Do you have an example/reference?

Reply:
Not using `on`, but by generating a `matrix`